### ISSUE TITLE
Restoring the correct apiserver extraArgs configuration will not cause KCP to automatically recovery

### ISSUE THREAD

**Issue by huaqing1994** [kubernetes-sigs/cluster-api#8691]
### What steps did you take and what happened?

1. added an invalid parameter to KCP apiserver extraArgs.
2. KCP created a new Machine, and its apiserver could not start.
3. KCP is stuck below：
```
E0517 06:59:23.719313       1 controller.go:329] "Reconciler error" err="failed to reconcile certificate expiry for Machine/huaqing-sks-test-ippool-1-controlplane-pdbr9: unable to get certificate expiry for kube-apiserver on Node/huaqing-sks-test-ippool-1-controlplane-5mzxf: TLS handshake with the kube-apiserver failed: EOF" controller="kubeadmcontrolplane" controllerGroup="controlplane.cluster.x-k8s.io" controllerKind="KubeadmControlPlane" KubeadmControlPlane="default/huaqing-sks-test-ippool-1-controlplane" namespace="default" name="huaqing-sks-test-ippool-1-controlplane" reconcileID=bd0673fc-137f-44df-90d0-f376282992b0
```
4. modify KCP to restore the original correct extraArgs.
5. but KCP is still stuck here, no automatic recovery.

### What did you expect to happen?

After step 4, KCP should automatically recovery.

### Cluster API version

v1.4.2

### Kubernetes version

1.24.8

### Anything else you would like to add?

It's work in CAPI v1.2.7. But not now.

This appears to be a bug in the "Detect Certificate Expiration" feature.
When there is an apiserver exception, `reconcileCertificateExpiries()` keeps returning error, preventing recovery.

Should `reconcileCertificateExpiries()` skip unhealthy apiservers?
I'm not sure if this is possible, but I would like KCP to automatically recovery like before.
https://github.com/kubernetes-sigs/cluster-api/blob/7b92ce4577e36adf013dc1dfd2b239fde4c36bc4/controlplane/kubeadm/internal/controllers/controller.go#L728-L746

### Label(s) to be applied

/kind bug

**Comment by killianmuldoon** [comment]
/triage accepted

I'll take a look at reproducing this today

**Comment by killianmuldoon** [comment]
@huaqing1994 - I took a look at this and it's definitely a bug IMO. It would be great if you could take a look at #8711 to review and ensure it fixes your issue.

**Comment by fabriziopandini** [comment]
I have commented on the PR but the TL;DR is that recovery from errors in CAPI is mostly driven by remediation, which requires MHC. Remediation already was capable to recover from the scenario above (error in retrieving the certificate expiration date).

However the error above was blocking rollouts, so the change proposed IMO makes sense

**Comment by huaqing1994** [comment]
@killianmuldoon  Thanks!
This PR does solve my problem!
